clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -184,6 +184,11 @@ export class CopybookDownloadService implements vscode.Disposable {
      */
     public async downloadCopybooks(cobolFileName: string, copybookNames: string[], quiet: boolean = true)
         : Promise<void> {
+        if (!ZoweVsCodeExtension.getZoweExplorerApi()) {
+            if (!quiet) vscode.window.showErrorMessage(ZOWE_EXT_MISSING_MSG);
```

Comment:
Good place.
In the beginning, I thought that it will be better also to check the settings like `cobol-lsp.cpy-manager.paths-dsn`, if it is specified then the user expects to work with Zowe. Otherwise don't show the error.
But we are out of quiet mode only in the case when the user manually hits "Resolve copybook" and expects that Zowe will work. Looks fine, but there is one interesting case:

We can reach copybooks in folders outside of the workspace and the plugin doesn't watch them. So, "Resolve copybook" is the only way for the user to rescan this folder. But we agreed that outside folders must not be changed.

Anyway, I think that your way is good. Maybe others can add something in the case I mention.

Reply:
For copybook-resolve, in folders outside workspace, it should be resolved by `copybookResolverURI.resolveCopybookURI` , and if it is not resolved then we try to download it and fails as zowe explorer is not present. Or am I missing something, or I am not able to understand the scenario altogether ( happens to me very often 😵‍💫 ).

Reply:
As I understood, by this line all the ways of the local resolving have been passed, so the outside-folder case should not be affected by these changes. I can say that this should work fine, just test it manually, please. I will also run the UI tests.

Reply:
I thought about corner cases.
We assume that if we are out of silent mode and don't have Zowe, we can warn the user that he needs the Zowe plugin. Is it always true? I mean can the user be in a case when he wants to work local only but see this Zowe warning?
When workspace folders are used for copybooks, the user typed "COPY FOO", but don't have the FOO copybook, he will see an error. After he creates the FOO copybook the error disappears on its own. So, no need to click "Resolve copybook" manually.
But when external folders are used the error keeps after the user added the FOO copybook. In this case, he must click "Resolve copybook", this is the only way.
This was my thought. And now I realize that after the external folder will have the FOO copybook the user will not see any error at all.

Reply:
This makes sense.
@zimlu02 any thoughts?

Reply:
But Anton, if we are talking about local non-workspace folders, we shouldn't mention ZE at all in error messages, right? There is no need for any connection to mainframe...  
Anyway, can we simply manually try to confirm how it behaves with this code? 
My take is that if the user adds the FOO external folder and hits 'resolve', it should only show the error if the path is not correct.

Reply:
I agree that we shouldn't mention ZE if the user didn't mean to use ZE.

In case when the user has "COPY FOO", then added FOO to external folder and hits 'resolve' everything will be fine. But if he had "COPY BAR" too and we can't find it locally, the plugin will show ZE error.

Reply:
So, does it make sense to avoid copybook download if ZE is not installed?